integration-tests/singleton_flush_test.go

```diff
@@ -38,81 +40,48 @@ func (suite *SingletonFlushTestSuite) SetupSuite() {
 
 }
 
+// Gets triggered before each test and runs the pre-conditions like setting up apisonator services
+// and generating envoy config from a template.
 func (suite *SingletonFlushTestSuite) BeforeTest(suiteName, testName string) {
 	if testName == "TestSingletonContainerFlush" {
-		configErr := configureSingletonFlush("ContainerLimit")
+		suite.ServiceID = uuid.NewString()
+		suite.ServiceToken = uuid.NewString()
+
+		// Configure the envoy template with ContainerLimit flush.
+		configErr := configureSingletonFlush("ContainerLimit", "100", "60", suite.ServiceID, suite.ServiceToken)
```

Comment:
It's just preference but I think if you are using `if testName == ...` then better add these conditions into the test as it is because now, readers have to be aware of two blocks for a single test i.e `BeforeTest` and the test function itself.

Reply:
okay will add the pre-conditions to the test itself.